src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/PEReaderExtensions.cs

```diff
@@ -70,14 +70,15 @@ private PEExportTable(PEReader peReader)
             }
 
             BlobReader namePointerReader = peImage.GetReader(peReader.GetOffset(namePointerRVA), sizeof(int) * namePointerCount);
+            StringBuilder nameBuilder = new StringBuilder();
```

Comment:
Same here.  If namePointerCount is 0, this will incur more allocation than previously.